dll/win32/shlwapi/utils.cpp

```diff
@@ -6,13 +6,31 @@
  */
 
 #define _ATL_NO_EXCEPTIONS
+
+/*
+ * HACK!  These functions are conflicting with <shobjidl.h> inline functions...
+ */
+#define IShellFolder_GetDisplayNameOf _disabled_IShellFolder_GetDisplayNameOf_
+#define IShellFolder_ParseDisplayName _disabled_IShellFolder_ParseDisplayName_
+#define IShellFolder_CompareIDs _disabled_IShellFolder_CompareIDs_
```

Comment:
Note that the `__IShellFolder_INTERFACE_DEFINED__` is defined when the shobjidl.h auto-generated header (from the IDL file) is included. I don't know whether this could be of any help for you.

Reply:
There are some necessary constants...